ui/src/app/workflows/components/workflows-toolbar/workflows-toolbar.tsx

```diff
@@ -47,11 +47,19 @@ export class WorkflowsToolbar extends React.Component<WorkflowsToolbarProps, {}>
     }
 
     private performActionOnSelectedWorkflows(ctx: any, title: string, action: WorkflowOperationAction): Promise<any> {
-        const confirmed = confirm(`Are you sure you want to ${title.toLowerCase()} all selected workflows?`);
         let deleteArchived = false;
+        const confirmed = confirm(`Are you sure you want to ${title.toLowerCase()} all selected workflows?`);
         if (confirmed) {
             if (title === 'DELETE') {
-                deleteArchived = confirm('Would you also want to delete them in the database?');
+                let hasArchived = false;
+                this.props.selectedWorkflows.forEach((wf: Workflow) => {
+                    if (!hasArchived && isArchivedWorkflow(wf)) {
+                        hasArchived = true;
+                    }
+                });
+                if (hasArchived) {
+                    deleteArchived = confirm('Would you also want to delete them from the Archived Workflows database?');
+                }
```

Comment:
```suggestion
                for (const entry of this.props.selectedWorkflows) {
                    if (isArchivedWorkflow(entry[1])) {
                        deleteArchived = confirm('Would you also want to delete them in the database?');
                        break
                    }
                }
```

ok that actually ended up being a decent simplification / reduction